### ISSUE TITLE
[Bug]: Noisy update cron job 

### ISSUE THREAD

**Issue by mikehearn** [netdata/netdata#12824]
### Bug description

After installing the netdata agent I get a noisy no-op apt-get output from cron every day. This is pointless, as nothing was actually changed. I disabled auto updates for now as I use unattended-upgrades anyway.

### Expected behavior

No spammy cron emails.

### Steps to reproduce

1. Install using kickstart script on Ubuntu.
2. Wait.

### Installation method

kickstart.sh

### System info

```shell
# uname -a; grep -HvE "^#|URL" /etc/*release
Linux hq.hydraulic.software 5.4.0-109-generic #123-Ubuntu SMP Fri Apr 8 09:10:54 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=20.04
/etc/lsb-release:DISTRIB_CODENAME=focal
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 20.04.4 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="20.04.4 LTS (Focal Fossa)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 20.04.4 LTS"
/etc/os-release:VERSION_ID="20.04"
/etc/os-release:VERSION_CODENAME=focal
/etc/os-release:UBUNTU_CODENAME=focal
```

### Netdata build info

```shell
# netdata -W buildinfo
Version: netdata v1.34.1
Configure options:  '--build=x86_64-linux-gnu' '--includedir=${prefix}/include' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--disable-silent-rules' '--libdir=${prefix}/lib/x86_64-linux-gnu' '--libexecdir=${prefix}/lib/x86_64-linux-gnu' '--disable-maintainer-mode' '--disable-dependency-tracking' '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libdir=/usr/lib' '--libexecdir=/usr/libexec' '--with-user=netdata' '--with-math' '--with-zlib' '--with-webdir=/var/lib/netdata/www' 'build_alias=x86_64-linux-gnu' 'CFLAGS=-g -O2 -fdebug-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -fdebug-prefix-map=/usr/src/netdata=. -fstack-protector-strong -Wformat -Werror=format-security'
Install type: binpkg-deb
    Binary architecture: x86_64
    Packaging distro:
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              YES
    ACLK Next Generation:       YES
    ACLK-NG New Cloud Protocol: YES
    ACLK Legacy:                NO
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         YES
Libraries:
    protobuf:                YES (system)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    YES
    EBPF:                    YES
    IPMI:                    YES
    NFACCT:                  YES
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: YES
```

**Comment by solracsf** [comment]
Same here, using `--static-only` option.

```
/etc/cron.daily/netdata-updater:

  ^
  |.-.   .-.   .-.   .-.   .  netdata
  |   '-'   '-'   '-'   '-'   real-time performance monitoring, done right!
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

  (C) Copyright 2017, Costa Tsaousis
  All rights reserved
  Released under GPL v3+

  You are about to install netdata to this system.
  netdata will be installed at:

                    /opt/netdata

  The following changes will be made to your system:

  # USERS / GROUPS
  User 'netdata' and group 'netdata' will be added, if not present.

  # LOGROTATE
  This file will be installed if logrotate is present.

   - /etc/logrotate.d/netdata

  # SYSTEM INIT
  This file will be installed if this system runs with systemd:

   - /lib/systemd/system/netdata.service

   or, for older CentOS, Debian/Ubuntu or OpenRC Gentoo:

   - /etc/init.d/netdata         will be created


  This package can also update a netdata installation that has been
  created with another version of it.

  Your netdata configuration will be retained.
  After installation, netdata will be (re-)started.

  netdata re-distributes a lot of open source software components.
  Check its full license at:
  https://github.com/netdata/netdata/blob/master/LICENSE
Creating directory /opt/netdata
Verifying archive integrity...     0%            9%           19%           28%           38%           47%           57%           67%           76%           86%           95%         100%   All good.
Uncompressing netdata, the real-time performance and health monitoring system     0%            9%           19%           28%           38%           47%           57%           67%           76%           86%           95%         100%  
 --- Attempt to create user/group netdata/netadata --- Group 'netdata' already exists.
User 'netdata' already exists.
 --- Add user netdata to required user groups --- Group 'docker' does not exist.
9
User 'netdata' is already in group 'nginx'.
Group 'varnish' does not exist.
Group 'haproxy' does not exist.
0
Adding netdata user to the adm group ...
[/opt/netdata]# usermod -a -G adm netdata  OK  ''

Group 'nsd' does not exist.
9
User 'netdata' is already in group 'proxy'.
Group 'squid' does not exist.
Group 'ceph' does not exist.
Group 'nobody' does not exist.
Group 'I2C' does not exist.
 --- Install logrotate configuration for netdata --- [/opt/netdata]# chmod 644 /etc/logrotate.d/netdata  OK  ''

 --- Telemetry configuration ---
You can opt out from anonymous statistics via the --disable-telemetry option, or by creating an empty file /opt/netdata/etc/netdata/.opt-out-from-anonymous-statistics 

 --- Install netdata at system init --- Installing systemd service...
[/opt/netdata]# cp system/netdata.service /lib/systemd/system/netdata.service
 OK  ''

[/opt/netdata]# systemctl daemon-reload  OK  ''

[/opt/netdata]# systemctl enable netdata  OK  ''

 --- Install (but not enable) netdata updater tool --- Unit /lib/systemd/system/netdata-updater.timer is masked, ignoring.
cat: /system/netdata-updater.timer: No such file or directory
cat: /system/netdata-updater.service: No such file or directory Update script is located at /opt/netdata/usr/libexec/netdata/netdata-updater.sh

 --- creating quick links ---
[/opt/netdata]# rm sbin
 OK  ''

[/opt/netdata]# ln -s bin sbin
 OK  ''

[/opt/netdata/usr]# rm bin
 OK  ''

[/opt/netdata/usr]# ln -s ../bin bin
 OK  ''

[/opt/netdata/usr]# rm sbin
 OK  ''

[/opt/netdata/usr]# ln -s ../bin sbin
 OK  ''

[/opt/netdata/usr]# rm local
 OK  ''

[/opt/netdata/usr]# ln -s . local
 OK  ''

[/opt/netdata]# rm netdata-configs
 OK  ''

[/opt/netdata]# ln -s etc/netdata netdata-configs  OK  ''

[/opt/netdata]# rm netdata-web-files
 OK  ''

[/opt/netdata]# ln -s usr/share/netdata/web netdata-web-files  OK  ''

[/opt/netdata]# rm netdata-plugins
 OK  ''

[/opt/netdata]# ln -s usr/libexec/netdata netdata-plugins  OK  ''

[/opt/netdata]# rm netdata-dbs
 OK  ''

[/opt/netdata]# ln -s var/lib/netdata netdata-dbs  OK  ''

[/opt/netdata]# rm netdata-metrics
 OK  ''

[/opt/netdata]# ln -s var/cache/netdata netdata-metrics  OK  ''

[/opt/netdata]# rm netdata-logs
 OK  ''

[/opt/netdata]# ln -s var/log/netdata netdata-logs  OK  ''

[/opt/netdata/etc/netdata]# rm orig
 OK  ''

[/opt/netdata/etc/netdata]# ln -s ../../usr/lib/netdata/conf.d orig  OK  ''

 --- fix permissions ---
[/opt/netdata]# chmod g+rx,o+rx /opt
 OK  ''

[/opt/netdata]# chown -R netdata:netdata /opt/netdata  OK  ''

 --- changing plugins ownership and setting setuid --- [/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/apps.plugin
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/apps.plugin
 OK  ''

[/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/ioping
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/ioping
 OK  ''

[/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/cgroup-network
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/cgroup-network
 OK  ''

[/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/ebpf.plugin
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/ebpf.plugin
 OK  ''

[/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/perf.plugin
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/perf.plugin
 OK  ''

[/opt/netdata]# chown root:netdata usr/libexec/netdata/plugins.d/slabinfo.plugin
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/slabinfo.plugin
 OK  ''

[/opt/netdata]# chown root:netdata bin/fping  OK  ''

[/opt/netdata]# chmod 4750 bin/fping
 OK  ''

Configure TLS certificate paths
Using /etc/ssl for TLS configuration and certificates Save install options
 --- starting netdata ---
 --- Restarting netdata instance --- 

Stopping all netdata threads
[/opt/netdata]# stop_all_netdata
 OK  ''

Starting netdata using command 'systemctl start netdata'
[/opt/netdata]# systemctl start netdata  OK  ''


  ^
  |.-.   .-.   .-.   .-.   .-.   .  netdata  .-.   .-.   .-.   .-.   .-.   .-
  |   '-'   '-'   '-'   '-'   '-'               '-'   '-'   '-'   '-'   '-'   
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

[/opt/netdata]# chmod 0644 /opt/netdata/etc/netdata/netdata.conf 
 OK  ''
```

**Comment by iigorkarpov** [comment]
@Ferroin @maneamarius 
We have to remove the symbolic link creation from installation/updating procedure

`/etc/cron.daily/netdata-updater -> /usr/libexec/netdata/netdata-updater.sh`

and create the new file in `/etc/cron.d/` containing something like this

```
MAILTO=
# run updater at 05:07 every night
7 5  * * * root /usr/libexec/netdata/netdata-updater.sh
```
In case if this file already exists, it should not be rewritten.

Corresponding changes in documentation also required.

**Comment by solracsf** [comment]
@iigorkarpov `--static-only` installs should also be handled.

`/etc/cron.daily/netdata-updater -> /opt/netdata/usr/libexec/netdata/netdata-updater.sh`

**Comment by PatTheMav** [comment]
Thank you for this issue, I also get a nightly mail with a whole lot of `apt` progress output for every server I have netdata running on.

I'd actually prefer if netdata uses `systemd` timers instead, but it looks like the `binpkg` variant does not install the necessary unit files, so even running `sudo /usr/libexec/netdata/netdata-updater.sh --enable-auto-updates systemd` fails with `Failed to enable unit: Unit file netdata-updater.timer does not exist.`.

**Comment by iigorkarpov** [comment]
@PatTheMav systemd timers do not provide an easy way to deliver email report from the job unfortunately. Probably this was the main reason for using cron. And if our priorities are changed, we probably could use it as well. At least on the systems with systemd service.

**Comment by Ferroin** [comment]
@iigorkarpov Could you please provide an actual explanation for your reasoning that we can’t use a symlink in `/etc/cron.daily` instead of just stating it as if it’s an indisputable fact.

---

Regarding the current handling of scheduling of automatic updates:

- We _currently_ support three different scheduling mechanisms:
    1. Symlinking the script in either `/etc/cron.daily` or `/etc/periodic/daily`.
    2. Using a systemd timer unit (which appears to be broken on at least some native package installs right now).
    3. Adding a drop-in crontab in `/etc/cron.d`.
- When enabling automatic updates, we automatically fall back in the order listed above if a given approach does not work. The order was chosen based on how many major distributions each approach works on (the symlink works almost everywhere, the systemd approach works on any systemd-based system, and the crontab approach only works on a handful of systems with a handful of cron implementations).
- Users can override the selection of auto-update mechanism on install by using the `--auto-update-type` option for the kickstart script (supported values are `interval`, `systemd`, and `crontab`, corresponding to the above list of options).
- With the latest nightlies and most recent stable release, users can also switch the mechanism used by manually running the updater script first with the `--disable-auto-updates` option, and then a second time with the `--enable-auto-updates` option (which takes the same values as the above mentioned `--auto-update-type` option for the kickstart script).

**Comment by iigorkarpov** [comment]
As you no doubt know, in a Unix/Linux type system, the possible solution is rarely the only one. This was the simplest solution that came to my mind when I saw the current way of calling the script by schedule. And flexible as well - the user could easily forbid sending any mail using the `MAILTO=` construct with an empty right hand side.
Later, when I became more familiar with the script itself, I saw that it also included my proposed `/etc/cron.d/` option and the timer option, which seemed more correct for a `systemd` system.
At the same time, I found that the very logic of using the above methods is broken. For example, there are no files concerning `systemd` in the packages.
So now it seems important to me to fix this package build error first. Most likely the problem will then go away by itself.
